### PR TITLE
fix: allow schema validation errors for jwts again

### PR DESCRIPTION
**Pull request by stevelikesmusic**
Closes: #N/A

### Description:
This PR improves JWT token validation for embedded dashboards by:

1. Updating the schema to properly handle `null` values for `allowedFilters` in dashboard filter interactivity options, which is needed for Python clients that serialize `None` to `null`.

2. Enhancing error handling in `decodeLightdashJwt` to log schema validation errors instead of throwing exceptions, reverting back to previous behavior. 

3. Improving error messages to be more specific about token schema validation errors.

4. Adding tests to verify that `null` values for `allowedFilters` are properly handled and that invalid payload structures are logged without throwing exceptions.

### CODE REVIEW

**Comment by stevelikesmusic** [comment]
* **#15619** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/15619?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/15619?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/common/src/ee/embed/index.ts [hunk 1]
```diff
@@ -101,7 +102,7 @@ type CommonEmbedJwtContent = {
     isPreview?: boolean;
     dashboardFiltersInteractivity?: {
         enabled: FilterInteractivityValues | boolean;
-        allowedFilters?: string[];
+        allowedFilters?: string[] | null;
```
Comment (rephus) [inline packages/common/src/ee/embed/index.ts:105]: so now we accept undefined | null | [] , right ?

File: packages/backend/src/auth/lightdashJwt.test.ts [hunk 2]
```diff
@@ -177,34 +194,26 @@ describe('JwtUtil', () => {
...
+    it('allows null for allowedFilters', () => {
+        const token = createTokenData({
+            dashboardFiltersInteractivity: {
+                allowedFilters: null,
+                enabled: false,
+            },
         });
+        const decoded = decodeLightdashJwt(token, encodedSecret);
+        expect(decoded).toBeTruthy();
+        expect(mockErrorLogger).not.toHaveBeenCalled();
```
Comment (rephus) [inline packages/backend/src/auth/lightdashJwt.test.ts:206]: nice

**Review by rephus — APPROVED** [review]

File: packages/common/src/ee/embed/index.ts (hunk 1 above)
Comment (stevelikesmusic) [reply packages/common/src/ee/embed/index.ts:105]: That's right. The problem was that python serializes `None` to `null` rather than `undefined`. In our docs, we tell Python users for set `allowedFilters: None`. So Python users have been always failing validation, but we haven't been enforcing the schema until I "fixed" it 😅
